posthog/api/error_tracking.py

```diff
@@ -452,13 +454,69 @@ def create(self, request, *args, **kwargs) -> Response:
-        # The rust HogVM expects a return statement, so we wrap the compiled filter expression in one
-        with_return = ast.ReturnStatement(expr=expr)
-        bytecode = create_bytecode(with_return).bytecode
-        validate_bytecode(bytecode)
-        return bytecode
+
+class ErrorTrackingGroupingRuleSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ErrorTrackingGroupingRule
+        fields = ["id", "filters"]
+        read_only_fields = ["team_id"]
+
+
+class ErrorTrackingGroupingRuleViewSet(TeamAndOrgViewSetMixin, viewsets.ModelViewSet):
+    scope_object = "error_tracking"
+    queryset = ErrorTrackingGroupingRule.objects.all()
+    serializer_class = ErrorTrackingGroupingRuleSerializer
+
+    def safely_get_queryset(self, queryset):
+        return queryset.filter(team_id=self.team.id)
+
+    def update(self, request, *args, **kwargs) -> Response:
+        grouping_rule = self.get_object()
+        assignee = request.data.get("assignee")
+        json_filters = request.data.get("filters")
+        description = request.data.get("description")
+
+        if json_filters:
+            parsed_filters = PropertyGroupFilterValue(**json_filters)
+            grouping_rule.filters = json_filters
+            grouping_rule.bytecode = generate_byte_code(self.team, parsed_filters)
+
+        if assignee:
+            grouping_rule.user_id = None if assignee["type"] != "user" else assignee["id"]
+            grouping_rule.user_group_id = None if assignee["type"] != "user_group" else assignee["id"]
+            grouping_rule.role_id = None if assignee["type"] != "role" else assignee["id"]
+
+        if description:
+            grouping_rule.description = description
+
+        grouping_rule.save()
+
+        return Response({"ok": True}, status=status.HTTP_204_NO_CONTENT)
+
+    def create(self, request, *args, **kwargs) -> Response:
+        json_filters = request.data.get("filters")
+        assignee = request.data.get("assignee", None)
+        description = request.data.get("description", None)
+
+        if not json_filters:
+            return Response({"error": "Filters are required"}, status=status.HTTP_400_BAD_REQUEST)
+
+        parsed_filters = PropertyGroupFilterValue(**json_filters)
+        bytecode = generate_byte_code(self.team, parsed_filters)
+
+        grouping_rule = ErrorTrackingGroupingRule.objects.create(
+            team=self.team,
+            filters=json_filters,
+            bytecode=bytecode,
+            order_key=0,
+            user_id=None if assignee["type"] != "user" else assignee["id"],
+            user_group_id=None if assignee["type"] != "user_group" else assignee["id"],
+            role_id=None if assignee["type"] != "role" else assignee["id"],
+            description=description,
+        )
+
+        serializer = ErrorTrackingGroupingRuleSerializer(grouping_rule)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
```

Comment:
Same as below but this all feels so similar that maybe it could be deduped in some way 🤔 Fine for now I guess